Add custom rules documentation (#27879)
DataDog/documentation: content/en/security/code_security/static_analysis/custom_rules/tutorial.md

## Step 2: Create a rule

First, create a rule named `tutorial-rule` and make sure you select the `Python` language.

### Step 2.1: Add an example

Add a code example to test the rule. At each modification or update of the rule, tree-sitter capture
or test code, Datadog will execute the rule against the code to have early feedback.

Enter the filename `test.py` and add the code below.

```python
foo(bar)
foo(baz)
```

### Step 2.2: Write a tree-sitter query

We then write a [tree-sitter query][2] that defines the nodes to capture in the code. In our case,
we want to capture a function call where the function name is `foo` and any argument is `bar`.

The tree-sitter query to do so is shown below, it has three captures:
 - `funcName` to capture the name of the function and run a predicate to check its name
 - `arg` to capture the name of the arguments
 - `func` to capture the function call with all arguments. This is used to report the violation in code.

```
(call
    function: (identifier) @funcName
    arguments: (argument_list
        (identifier) @arg
    )
    (#eq? @funcName "foo")
    (#eq? @arg "bar")
)@func
```

### Step 2.3: Write JavaScript rule code

> Finally, we write the JavaScript code to report the violation. First, we fetch the captures (`func` and `funcName`).
> Then, we check if the name of the function is different from `foo` and return otherwise. Finally, we report a violation.

Reviewer: ```suggestion
Write the JavaScript code to report the violation. The code first fetch the captures (`func` and `funcName`), and then checks if the name of the function is different from `foo` and returns if true. Lastly, it reports a violation.
```

We support the following severity: `ERROR`, `WARNING`, `NOTICE` and `INFO`.
We support the following categories: `BEST_PRACTICES`, `CODE_STYLE`, `ERROR_PRONE`, `PERFORMANCE` and `SECURITY`.

```javascript
function visit(query, filename, code) {
  // Get the func captures so that we can know the line/col where to put a violation
  const func = query.captures["func"];

  // Get the funcname to later get the name of the function
  const funcNameNode = query.captures["funcName"];

  // Check if the name of the function is not foo. This is already done in the tree-sitter query
  // and here only to show how to use getCodeForNode
  const funcNameFromCode = getCodeForNode(funcNameNode, code);
  if (funcNameFromCode !== "foo") {
    return;
  }

  // Report a violation
  addError(
    buildError(
      func.start.line, func.start.col,
      func.end.line, func.end.col,
      "do not use bar as argument",
      "WARNING",
      "BEST_PRACTICES"
    )
  )
}
```

{{< img src="/security/code_security/custom_rule_tutorial_rule_created.png" alt="Rule created" style="width:100%;" >}}

## Step 3: Use the rule

> To use the rule, either:
>  - create a `static-analysis.datadog.yaml` file at the root of your repository with the ruleset
>  - add the rule in [your settings][3], either for the org-wide or repo-level configuration.

Reviewer: ```suggestion
To use the rule, do one of the following:
 - Create a `static-analysis.datadog.yaml` file at the root of your repository with the ruleset.
 - Add the rule in [your settings][3], either for the org-wide or repo-level configuration.
```
